src/project-generator.ts

```diff
@@ -51,266 +39,97 @@ export class ProjectGenerator {
-            } else {
-                throw error;
-            }
-        }
-    }
-
-    private async updateTree(sha1: string, sha2: string): Promise<any> {
-        try {
-            const baseTreeSha = await this.getBaseTreeSha();
-            const newTreeSha = await this.createNewTreeSha(sha1, sha2, baseTreeSha);
-            const mainBranchSha = await this.getMainBranchSha();
-            const newCommitSha = await this.createCommitSha(mainBranchSha, newTreeSha);
-            await delay(MS_TO_WAIT_FOR_GITHUB);
-            await this.setDefaultWorkflowPermissionToWrite();
-            await this.enableWorkflows(true);
-            await this.updateMainBranchSha(newCommitSha);
-            return StatusCodes.OK;
-        } catch (error) {
-            console.log(error);
-        }
+        await this.gitRequestHandler.updateTree(appManifestBlobSha, mainPyBlobSha, vspecJsonBlobSha);
+        return StatusCodes.OK;
     }
 
-    private async updateContent(appName: string, codeSnippet: string): Promise<any> {
-        const appManifestContentData = await this.getFileContentData('AppManifest');
-        let decodedContent = JSON.parse(
-            Buffer.from(appManifestContentData, CONTENT_ENCODINGS.base64 as BufferEncoding).toString(
-                CONTENT_ENCODINGS.utf8 as BufferEncoding
-            )
-        );
-        decodedContent[0].Name = appName.toLowerCase();
-        const updateContentString = JSON.stringify(decodedContent, null, 4);
-        const encodedUpdateContent = Buffer.from(updateContentString, CONTENT_ENCODINGS.utf8 as BufferEncoding).toString(
-            CONTENT_ENCODINGS.base64 as BufferEncoding
-        );
-        const appManifestBlobSha = await this.createBlob(encodedUpdateContent);
-        const mainPyContentData = await this.getFileContentData('main');
-        const convertedMainPy = this.codeConverter.convertMainPy(mainPyContentData, codeSnippet, appName);
-        const mainPyBlobSha = await this.createBlob(convertedMainPy);
-        await this.updateTree(appManifestBlobSha, mainPyBlobSha);
-    }
+    private async getNewAppManifestSha(appName: string, vspec: string): Promise<string> {
+        const appManifestContentData = await this.gitRequestHandler.getFileContentData('AppManifest');
+        let decodedAppManifestContent = JSON.parse(decode(appManifestContentData));
+        decodedAppManifestContent[0].Name = appName.toLowerCase();
 
-    private async getFileContentData(file: string): Promise<string> {
-        let fileContentResponse;
-        if (file === 'AppManifest') {
-            fileContentResponse = await axios.get(`${this.repositoryPath}/contents/app/AppManifest.json`, this.requestConfig);
-        } else if (file === 'main') {
-            fileContentResponse = await axios.get(`${this.repositoryPath}/contents/app/src/main.py`, this.requestConfig);
+        if (this.vspecIsPayload(vspec)) {
+            decodedAppManifestContent[0].vspec = './app/vspec.json';
```

Comment:
Careful: Location of the `vspec` attribute in `AppManifest` has not been decided, yet!

Reply:
The information whether vspec is a payload or not is available in all calling methods. I recommend we do not forget this information ant try to retrieve it with an additional method.